lib/livebook/intellisense.ex

```diff
@@ -207,7 +221,96 @@ defmodule Livebook.Intellisense do
       insert_text: name
     }
 
-  @ordered_kinds [:field, :variable, :module, :struct, :interface, :function, :type]
+  defp keyword_macro?(name) do
+    def? = name |> Atom.to_string() |> String.starts_with?("def")
+
+    def? or
+      name in [
+        :alias,
+        :import,
+        :require,
+        :use,
+        :case,
+        :cond,
+        :for,
+        :if,
+        :quote,
+        :raise,
+        :receive,
+        :throw,
+        :try,
+        :unless,
+        :with
+      ]
+  end
+
+  defp env_macro?(name) do
+    name in [:__ENV__, :__MODULE__, :__DIR__, :__STACKTRACE__, :__CALLER__]
+  end
+
+  defp extra_completion_items(hint) do
+    items = [
+      %{
+        label: "do",
+        kind: :keyword,
+        detail: "do-end block",
+        documentation: nil,
+        insert_text: "do\n  $0\nend"
+      },
+      %{
+        label: "end",
+        kind: :keyword,
+        detail: "do-end block",
+        documentation: nil,
+        insert_text: "end"
+      },
+      %{
+        label: "fn",
+        kind: :keyword,
+        detail: "anonymous function",
+        documentation: nil,
+        insert_text: "fn"
+      },
```

Comment:
I just realized that `fn` shows up as a special form so we can likely remove this.

Reply:
Ah right, we should make sure fn inserts a space and not parentheses and we should be good.

I was tempted to do `fn $1 -> $0 end`, but in the end it's probably a flow killer anyway ^^